File: frontend/apis/jobProcessingApi/Models/ResolvedJobResponse.ts
import {UserResponse} from './UserResponse'
import {Price} from './Price'
import {JobUrgency} from './JobUrgency'
import {JobDurationType} from './JobDurationType'
import {Address} from './Address'
export interface ResolvedJobResponse {
    id?: string | undefined;
    name?: string | undefined;
    description?: string | undefined;
    highlightedDescription?: string | undefined;
    numberOfEmployeesRequired?: number;
    location?: Address;
    jobDurationType?: JobDurationType;
    price?: Price;
    urgency?: JobUrgency;
    createDate?: string;
    categoryId?: number;
    categoryName?: string | undefined;
    employer?: UserResponse;
}

File: backend/EasyJob/Models/Job/ResolvedJobResponse.cs
//using Google.Protobuf.WellKnownTypes;
//using JobProcessing.Application;
//using Profile;
//using System;
//using System.Collections.Generic;
//using System.Linq;
//using System.Threading.Tasks;

using JobProcessing;
using Profile;
using System;

namespace EasyJob.Models.Job
{
    public class ResolvedJobResponse
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string HighlightedDescription { get; set; }
        public int NumberOfEmployeesRequired { get; set; }
        public Address Location { get; set; }
        public JobDurationType JobDurationType { get; set; }
        public Price Price { get; set; }
        public JobUrgency Urgency { get; set; }
        public DateTimeOffset CreateDate { get; set; }
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public UserResponse Employer { get; set; }
    }
}
